clippy_lints/src/attrs/repr_attributes.rs

```diff
@@ -0,0 +1,40 @@
+use rustc_ast::Attribute;
+use rustc_lint::LateContext;
+use rustc_span::{sym, Span};
+
+use clippy_utils::diagnostics::span_lint_and_then;
+
+use super::REPR_PACKED_WITHOUT_ABI;
+
+pub(super) fn check(cx: &LateContext<'_>, item_span: Span, attrs: &[Attribute]) {
+    check_packed(cx, item_span, attrs);
+}
+
+fn check_packed(cx: &LateContext<'_>, item_span: Span, attrs: &[Attribute]) {
+    if let Some(items) = attrs.iter().find_map(|attr| {
+        if attr.ident().is_some_and(|ident| matches!(ident.name, sym::repr)) {
+            attr.meta_item_list()
+        } else {
+            None
+        }
+    }) && let Some(packed) = items
+        .iter()
+        .find(|item| item.ident().is_some_and(|ident| matches!(ident.name, sym::packed)))
+        && !items.iter().any(|item| {
+            item.ident()
+                .is_some_and(|ident| matches!(ident.name, sym::C | sym::Rust))
+        })
+    {
+        span_lint_and_then(
+            cx,
+            REPR_PACKED_WITHOUT_ABI,
+            item_span,
+            "item uses `packed` representation without ABI-qualification",
+            |diag| {
+                diag.warn("unqualified `#[repr(packed)]` defaults to `#[repr(Rust, packed)]`, which has no stable ABI")
+                    .help("qualify the desired ABI explicity via `#[repr(C, packed)]` or `#[repr(Rust, packed)]`")
```

Comment:
Sorry for the wait!

[Before 1.74](https://github.com/rust-lang/rust/pull/114201/) you couldn't write `repr(Rust)`, adding an MSRV check is an option but I think leaving it out is reasonable too since you could just disable the lint if you want `repr(Rust, packed)` and are targeting older versions

I'll leave that choice up to you, the rest looks great

Reply:
Force-pushed a rebase. I've added a msrv filter, although I'm not sure given the docs if `msrv_aliases!` is supposed to only name actual feature-names as used in rustc; as far as I can see, `#[repr(Rust)]` didn't get a name, so I made something up. Bumped to 1.84. Had to rebase beyond 63d0ba9, and I'm unsure if `PostExpansionEarlyAttributes` is where this lint should live according to 63d0ba9's intent.

Reply:
It should be fine to go in `Attributes`, the late pass

MSRV name is 👍, just has to be understandable

Reply:
Moved it over